src/types.ts

```diff
@@ -209,3 +209,10 @@ export interface JWK {
 
   [propName: string]: unknown;
 }
+
+export interface GenericMetadata {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick(naming):**  It looks to me this struct matches a _trait‏_ (_à la_ Rust) or a portion of a bigger structure. I would use a name that makes it clear and is JWK-oriented, too. Something like `WithJWKS`.